app/src/main/java/zegreanu/cristi/geoquiz/QuizActivity.kt

```diff
@@ -95,12 +124,23 @@ class QuizActivity : AppCompatActivity() {
     private fun checkAnswer(userPressedTrue: Boolean) {
         val answerIsTrue = questionBank[currentIndex].answerTrue
         val messageResId =
-            if (userPressedTrue == answerIsTrue) R.string.correct_toast else R.string.incorrect_toast
+            when {
+                isCheater -> R.string.judgment_toast
+                userPressedTrue == answerIsTrue -> R.string.correct_toast
+                else -> R.string.incorrect_toast
+            }
 
         Toast.makeText(
             this@QuizActivity,
             messageResId,
             Toast.LENGTH_SHORT
         ).show()
     }
+
+    private fun newIntent(packageContext: Context, answerIsTrue: Boolean): Intent {
```

Comment:
Normally in Java this would bee a static method on the "destination" activity class. I think it can be the same in Android.

This static method helps reusing the intent creation code whenever we want to navigate to the destination activity.

You can find an example here - https://gist.github.com/rivancic/35a36ce43feb0bb713b2

I would stick with the `newIntent` naming.

Reply:
This also allows you to move the `EXTRA_ANSWER_IS_TRUE` constant and have it defined in a single place.

Reply:
Hmm - it seems in Kotlin you might be able to use something like https://medium.com/@passsy/starting-activities-with-kotlin-my-journey-8b7307f1e460